Sources/MapboxMaps/Foundation/BaseMapView.swift

```diff
@@ -222,21 +223,26 @@ open class BaseMapView: UIView, MapClient, MBMMetalViewProvider, CameraViewDeleg
             }
         }
 
-        self.resourceOptions = resourceOptions
+        self.resourceOptions = mapInitOptions.resourceOptions
         observerConcrete = ObserverConcrete()
 
-        let size = MapboxCoreMaps.Size(width: Float(frame.width), height: Float(frame.height))
-
-        let mapOptions = MapboxCoreMaps.MapOptions(__contextMode: nil,
-                                                   constrainMode: nil,
-                                                   viewportMode: nil,
-                                                   orientation: nil,
-                                                   crossSourceCollisions: nil,
-                                                   size: size,
-                                                   pixelRatio: Float(UIScreen.main.scale),
-                                                   glyphsRasterizationOptions: glyphsRasterizationOptions)
-
-        __map = try! Map(client: self, mapOptions: mapOptions, resourceOptions: resourceOptions)
+        let resolvedMapOptions: MapOptions
+
+        if mapInitOptions.mapOptions.size == nil {
+            // Update using the view's size
+            let other = mapInitOptions.mapOptions
+            resolvedMapOptions = MapOptions(__contextMode: other.__contextMode,
+                                            constrainMode: other.__constrainMode,
+                                            viewportMode: other.__viewportMode,
+                                            orientation: other.__orientation,
+                                            crossSourceCollisions: other.__crossSourceCollisions,
```

Comment:
do we need swift refinements for these MapOptions properties and this initializer?

Reply:
Maybe. I have another discussion going on about this elsewhere.

Reply:
Will ticket up resolving this & `mapboxAccessToken`